### PR TITLE
1700 bug es sink doc dont write pk

### LINKED ISSUE #1700: Bug: Es sink doc don't write pk
### Where's the bug?

https://docs.risingwave.com/docs/current/sink-to-elasticsearch/

### Describe the bug

Our sink's primary_key needs to be specified manually in the with option, not inherited from mv.And please add primary_key to the with menu.

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [two major changes: ]
  - [1. move parameter `type` and add some notes about it.]
  - [2. add `primary_key` parameter and add some notes about it.]

- **Notes**

  - [ Click [here](https://pr-1760.d2fbku9n2b6wde.amplifyapp.com/docs/dev/sink-to-elasticsearch/) to review.]

- **Related code PR**

  - [ https://github.com/risingwavelabs/risingwave/pull/14273]

- **Related doc issue**
  
  Resolves [ https://github.com/risingwavelabs/risingwave-docs/issues/1706#event-11367893646 ]
  Resolves [https://github.com/risingwavelabs/risingwave-docs/issues/1700#event-11451480039]
Resolves [https://github.com/risingwavelabs/risingwave-docs/issues/1693]

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/guides/sink-to-elasticsearch.md [hunk 1]
```diff
@@ -51,16 +51,26 @@ WITH (
...
+For versions under 8.x, there was once a parameter `type`. In Elasticsearch 6.x, users could directly set the type, but starting from 7.x, it is set to not recommended and the default value is unified to '_doc.' In version 8.x, the type has been completely removed. See [Elasticsearch's official documentation](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/removal-of-types.html) for more details.
+
+So, if you are using Elasticsearch 7.x, we set it to the Elasticsearch official default recommended value, which is '_doc'. If you are using Elasticsearch 8.x, this parameter has been removed by the Elasticsearch official, so no setting is required.
+:::
+
+### Notes about primary keys and Elasticsearch IDs
+
+The Elasticsearch sink defaults to the `upsert` sink type. It does not support the `append-only` sink type.
+
+If the sink_from object has primary keys, please specify them via the `primary_key` parameter. RisingWave will combine multiple primary key values into a single string with the delimiter you set, and use it as the Elasticsearch ID.
```
Comment (xxhZs) [inline docs/guides/sink-to-elasticsearch.md]: “If the sink_from object has primary keys” => modify it like "If you need to customize the ID"

File: docs/guides/sink-to-elasticsearch.md [hunk 2]
```diff
@@ -51,16 +51,26 @@ WITH (
...
 
-If the sink has a primary key (normally it is inherited from a materialized view), RisingWave will use the primary key as the Elasticsearch ID.
-If the sink doesn't have a primary key (in the case that the materialized view is append-only), RisingWave will use the first column in the sink definition as the Elasticsearch ID.
+If the sink_from object doesn't have a primary key, RisingWave will use the first column in the sink definition as the Elasticsearch ID.
```
Comment (xxhZs) [inline docs/guides/sink-to-elasticsearch.md]: ditto

**Review by xxhZs — APPROVED** [review]
lgtm

**Review by CharlieSYH — APPROVED** [review]

File: docs/guides/sink-to-elasticsearch.md [hunk 3]
```diff
@@ -92,6 +92,12 @@ ElasticSearch uses a mechanism called [dynamic field mapping](https://www.elasti
 |timestamp without time zone | text|
 |timestamp with time zone |text|
 |interval |text|
-|struct |Not supported|
+|struct |struct|
 |array |array|
-|JSONB |text|
+|JSONB|struct (RisingWave's Elasticsearch sink will send JSONB as a JSON string, and ElasticSearch will convert it into a struct structure)|
```
Comment (WanYixian) [inline docs/guides/sink-to-elasticsearch.md]: and **Elasticsearch** will convert it into a struct structure, lower case of "s" here
Others LGTM

**Review by WanYixian — APPROVED** [review]

**Comment by ShanlanLi** [comment]
This pr also resolves [https://github.com/risingwavelabs/risingwave-docs/issues/1693].

**Pull request merged**